packages/react/src/components/Pagination/index.test.tsx

```diff
@@ -0,0 +1,273 @@
+import React, { useState } from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import Pagination, { usePagination } from './';
+import axe from '../../axe';
+
+test('should set `thin` variant when thin prop is provided', () => {
+  render(<Pagination totalItems={18} currentPage={1} thin />);
+  expect(screen.getByRole('list').parentElement).toHaveClass(
+    'Pagination--thin'
+  );
+});
+
+test('should disable first/prev page buttons', () => {
+  const onFirstPageClick = jest.fn();
+  const onPreviousPageClick = jest.fn();
+
+  render(
+    <Pagination
+      totalItems={18}
+      currentPage={1}
+      onFirstPageClick={onFirstPageClick}
+      onPreviousPageClick={onPreviousPageClick}
+    />
+  );
+
+  expect(screen.getAllByRole('button')[0]).toHaveAttribute(
+    'aria-disabled',
+    'true'
+  );
+  expect(screen.getAllByRole('button')[1]).toHaveAttribute(
+    'aria-disabled',
+    'true'
+  );
+  expect(screen.getAllByRole('button')[2]).toHaveAttribute(
+    'aria-disabled',
+    'false'
+  );
+  expect(screen.getAllByRole('button')[3]).toHaveAttribute(
+    'aria-disabled',
+    'false'
+  );
+
+  fireEvent.click(screen.getAllByRole('button')[0]);
+  fireEvent.click(screen.getAllByRole('button')[1]);
+
+  expect(onFirstPageClick).not.toHaveBeenCalled();
+  expect(onPreviousPageClick).not.toHaveBeenCalled();
+});
+
+test('should disable last/next page buttons', () => {
+  const onNextPageClick = jest.fn();
+  const onLastPageClick = jest.fn();
+
+  render(
+    <Pagination
+      totalItems={18}
+      currentPage={2}
+      onNextPageClick={onNextPageClick}
+      onLastPageClick={onLastPageClick}
+    />
+  );
+
+  expect(screen.getAllByRole('button')[0]).toHaveAttribute(
+    'aria-disabled',
+    'false'
+  );
+  expect(screen.getAllByRole('button')[1]).toHaveAttribute(
+    'aria-disabled',
+    'false'
+  );
+  expect(screen.getAllByRole('button')[2]).toHaveAttribute(
+    'aria-disabled',
+    'true'
+  );
+  expect(screen.getAllByRole('button')[3]).toHaveAttribute(
+    'aria-disabled',
+    'true'
+  );
+
+  fireEvent.click(screen.getAllByRole('button')[2]);
+  fireEvent.click(screen.getAllByRole('button')[3]);
+
+  expect(onNextPageClick).not.toHaveBeenCalled();
+  expect(onLastPageClick).not.toHaveBeenCalled();
+});
+
+test('should support custom status label text', () => {
+  render(
+    <Pagination totalItems={18} statusLabel={<div id="foo">hello world</div>} />
+  );
+
+  expect(screen.getByText('hello world')).toHaveAttribute('id', 'foo');
```

Comment:
The one change I would make here is to do what you did below to check the log role:

```tsx
expect(screen.getByRole('log')).toHaveTextContent('Showing 35 to 51 of 500');
```